src/ingest/SpecRepo.cpp

```diff
@@ -158,8 +158,18 @@ void SpecRepo::genPatternSpec(long start,
   for (long i = start; i >= 0; i--) {
     auto str = pathTemplate;
     const auto watermark = now - i * curUnitInSeconds;
+
     const auto patternWithBracket = fmt::format("{{{0}}}", curPatternStr);
-    const auto pos = pathTemplate.find(patternWithBracket);
+    const auto opos = pathTemplate.find(patternWithBracket);
+
+    // uppercase pattern string
+    std::string upperCurPatternStr;
+    transform(curPatternStr.begin(), curPatternStr.end(), std::back_inserter(upperCurPatternStr), toupper);
+    const auto upperPatternWithBracket = fmt::format("{{{0}}}", upperCurPatternStr);
```

Comment:
slightly better in perf though it may not matter, why not one line

`std::transform(patternWithBracket.begin(), patternWithBracket.end(), patternWithBracket.begin(), std::toupper)`

make the same string as upper case in-place since you use it just once.

Reply:
good catch, fixed.